### DISCUSSION TITLE
Can't change default image sampler - DefaultImageSampler tuple field is pub(crate)

### DISCUSSION THREAD

**Discussion by sarkahn** [bevyengine/bevy#5046] [Q&A] [answered]
## Bevy version

main

## What went wrong

We now have the [DefaultImageSampler](https://github.com/bevyengine/bevy/blob/caa61c5fb7990a816766d0797a81a8a9ac07b3e8/crates/bevy_render/src/texture/image.rs#L152) resource, which I would have thought is a way for users to specify a different default sampler for new sprites. However because the tuple field is `pub(crate)` we have no way to actually change the resource externally. 

If we're not meant to change this resource why is it publicly visible?

**Comment by DJMcNab** [comment, marked as answer]
You should set the `default_sampler` field on [`ImageSettings`](http://dev-docs.bevyengine.org/bevy/render/texture/struct.ImageSettings.html).

The `DefaultImageSampler` gets set based on that:

https://github.com/bevyengine/bevy/blob/8b27124a801d83a1e92f1136ebd2bc9752c3e9d7/crates/bevy_render/src/texture/mod.rs#L73-L76

Notice that `ImageSettings` is a main world resource, whereas `DefaultImageSampler` is a render world resource.

**Reply by sarkahn** [reply]
Great, thanks! I'm wondering if the documentation for DefaultImageSampler should say something to that effect?

**Reply by sarkahn** [reply]
Also the documentation for [ImageSampler](https://github.com/bevyengine/bevy/blob/caa61c5fb7990a816766d0797a81a8a9ac07b3e8/crates/bevy_render/src/texture/image.rs#L114-L117) should probably reflect this as well. It mentions `DefaultImageSampler` but no mention of `ImageSettings` - the actual resource used to change the default.

**Reply by DJMcNab** [reply]
That would be because the change to `DefaultImageSampler` was one of @cart's 'drive-by' changes, which didn't get reviewed in light of the rest of the PR. A fix would be welcome.
